src/fondue/Fondue.js

```diff
@@ -760,7 +777,45 @@ export default class Fondue {
 				});
 			}
 
-			return currentAllGlyphs;
+			// Chained context substitution
+			if (lookup.lookupType === 6) {
+				lookup.subtableOffsets.forEach((_, i) => {
+					let subtable = lookup.getSubTable(i);
+
+					if (subtable.inputGlyphCount > 0) {
+						subtable.inputCoverageOffsets.forEach((offset) => {
+							const coverage = subtable.getCoverageFromOffset(
+								offset
+							);
+							parsedLookup["input"][i] = glyphToLetter(coverage);
+						});
+					}
+
+					if (subtable.backtrackGlyphCount > 0) {
+						subtable.backtrackCoverageOffsets.forEach((offset) => {
+							const coverage = subtable.getCoverageFromOffset(
```

Comment:
`getCoverageFromOffset` is performed for input, backtrack and lookahead. I think this could be one only once? Not sure if it's an expensive operation or not.

Reply:
Wowzers!! I was staring at this code for so long, I didn't see the obvious. Great simplification, thanks!